packages/@aws-cdk/aws-appmesh/lib/tls-certificate.ts

```diff
@@ -0,0 +1,192 @@
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import { CfnVirtualGateway, CfnVirtualNode } from './appmesh.generated';
+
+/**
+ * Enum of supported TLS modes
+ */
+export enum TlsMode {
+  /**
+   * Only accept encrypted traffic
+   */
+  STRICT = 'STRICT',
+
+  /**
+   * Accept encrypted and plaintext traffic.
+   */
+  PERMISSIVE = 'PERMISSIVE',
+
+  /**
+   * TLS is disabled, only accept plaintext traffic.
+   */
+  DISABLED = 'DISABLED',
+}
+
+/**
+ * Redundant API shapes are represented here.
+ */
+export interface TlsCertificateConfig {
+  /**
+   * The CFN shape for a virtual gateway listener TLS certificate
+   */
+  readonly virtualGatewayListenerTlsCertificate: CfnVirtualGateway.VirtualGatewayListenerTlsCertificateProperty,
+
+  /**
+   * The CFN shape for a virtual node listener TLS certificate
+   */
+  readonly virtualNodeListenerTlsCertificate: CfnVirtualNode.ListenerTlsCertificateProperty,
+}
+
+/**
+ * Represents a TLS certificate
+ */
+export abstract class TlsCertificate {
+  /**
+   * Returns an File TLS Certificate
+   */
+  public static file(props: FileCertificateOptions): TlsCertificate {
+    return new FileTlsCertificate(props);
+  }
+
+  /**
+   * Returns an ACM TLS Certificate
+   */
+  public static acm(props: AcmCertificateOptions): TlsCertificate {
+    return new AcmTlsCertificate(props);
+  }
+
+  /**
+   * The TLS mode.
+   *
+   * @default - none
+   */
+  readonly abstract tlsMode: TlsMode;
```

Comment:
Don't expose this, I don't think there's any reason to.

Reply:
`tlsMode` is referenced in listeners `renderTls()` function. I prefer setting it here because we might add fields to the `tls` shape that will depend on fields not in `tls-certificate`. However if you have a reason for not exposing this I think I can refactor it in a future proof way.

```typescript
function renderTls(scope: cdk.Construct, tlsCertificate: TlsCertificate): CfnVirtualGateway.VirtualGatewayListenerTlsProperty {
  return {
    certificate: tlsCertificate.bind(scope).tlsCertificate,
    mode: tlsCertificate.tlsMode.toString(),
  };
}
```